packages/google_mlkit_commons/android/src/main/java/com/google_mlkit_commons/InputImageConverter.java

```diff
@@ -39,16 +40,28 @@ public static InputImage getInputImageFromData(Map<String, Object> imageData,
                     Map<String, Object> metaData = (Map<String, Object>) imageData.get("metadata");
 
                     assert metaData != null;
-                    inputImage = InputImage.fromByteArray((byte[]) Objects.requireNonNull(imageData.get("bytes")),
-                            Double.valueOf(Objects.requireNonNull(metaData.get("width")).toString()).intValue(),
-                            Double.valueOf(Objects.requireNonNull(metaData.get("height")).toString()).intValue(),
-                            Integer.parseInt(Objects.requireNonNull(metaData.get("rotation")).toString()),
-                            Integer.parseInt(Objects.requireNonNull(metaData.get("image_format")).toString()));
-                    return inputImage;
+                    byte[] data = (byte[]) Objects.requireNonNull(imageData.get("bytes"));
+                    int imageFormat = Integer.parseInt(Objects.requireNonNull(metaData.get("image_format")).toString());
+                    int rotationDegrees = Integer.parseInt(Objects.requireNonNull(metaData.get("rotation")).toString());
+                    int width = Double.valueOf(Objects.requireNonNull(metaData.get("width")).toString()).intValue();
+                    int height = Double.valueOf(Objects.requireNonNull(metaData.get("height")).toString()).intValue();
+                    if (imageFormat == ImageFormat.YUV_420_888) {
```

Comment:
could you change the conditional to check that imageFormat is he supported formats are [IMAGE_FORMAT_NV21](https://developers.google.com/android/reference/com/google/mlkit/vision/common/InputImage#IMAGE_FORMAT_NV21) and [IMAGE_FORMAT_YV1](https://developers.google.com/android/reference/com/google/mlkit/vision/common/InputImage#IMAGE_FORMAT_YV12), which are the only supported for this method:

https://developers.google.com/android/reference/com/google/mlkit/vision/common/InputImage#public-static-inputimage-frombytebuffer-bytebuffer-bytebuffer,-int-width,-int-height,-int-rotationdegrees,-int-format